src/test/java/DirectionsServletTest.java

```diff
@@ -0,0 +1,175 @@
+          "[DirectionsLeg: \"Kitchener, ON, Canada\" -> \"Waterloo, ON, Canada\" (43.45184990,-80.49313410 -> 43.46430180,-80.52042120), duration=8 mins, distance=3.0 km: 5 steps]",
+          "[DirectionsLeg: \"Waterloo, ON, Canada\" -> \"Windsor, ON, Canada\" (43.46430180,-80.52042120 -> 42.31486680,-83.03656800), duration=2 hours 55 mins, distance=291 km: 18 steps]",
+          "[DirectionsLeg: \"Windsor, ON, Canada\" -> \"Montreal, QC, Canada\" (42.31486680,-83.03656800 -> 45.50171230,-73.56721840), duration=8 hours 42 mins, distance=897 km: 21 steps]");
+  private static final List<String> montrealToWaterlooNoWaypoints =
+      Arrays.asList(
+          "[DirectionsLeg: \"Montreal, QC, Canada\" -> \"Waterloo, ON, Canada\" (45.50171230,-73.56721840 -> 43.46430180,-80.52042120), duration=6 hours 11 mins, distance=638 km: 22 steps]");
+  private static final String WINDSOR_TO_MONTREAL_WITH_WAYPOINTS_JSON =
+      "[\"[DirectionsLeg: \"Windsor, ON, Canada\" -\u003e \"Waterloo, ON, Canada\" (42.31486680,-83.03656800 -\u003e 43.46430180,-80.52042120), duration\u003d2 hours 58 mins, distance\u003d293 km: 16 steps]\",\"[DirectionsLeg: \"Waterloo, ON, Canada\" -\u003e \"Markham, ON, Canada\" (43.46430180,-80.52042120 -\u003e 43.85644940,-79.33771690), duration\u003d1 hour 14 mins, distance\u003d123 km: 15 steps]\",\"[DirectionsLeg: \"Markham, ON, Canada\" -\u003e \"Quebec City, QC, Canada\" (43.85644940,-79.33771690 -\u003e 46.81380600,-71.20822600), duration\u003d7 hours 42 mins, distance\u003d790 km: 30 steps]\",\"[DirectionsLeg: \"Quebec City, QC, Canada\" -\u003e \"Montreal, QC, Canada\" (46.81380600,-71.20822600 -\u003e 45.50171230,-73.56721840), duration\u003d2 hours 46 mins, distance\u003d253 km: 29 steps]\"]";
+  private static final String MONTREAL_TO_MONTREAL_WITH_WAYPOINTS_JSON =
+      "[\"[DirectionsLeg: \"Montreal, QC, Canada\" -\u003e \"Kitchener, ON, Canada\" (45.50171230,-73.56721840 -\u003e 43.45184990,-80.49313410), duration\u003d6 hours 10 mins, distance\u003d632 km: 16 steps]\",\"[DirectionsLeg: \"Kitchener, ON, Canada\" -\u003e \"Waterloo, ON, Canada\" (43.45184990,-80.49313410 -\u003e 43.46430180,-80.52042120), duration\u003d8 mins, distance\u003d3.0 km: 5 steps]\",\"[DirectionsLeg: \"Waterloo, ON, Canada\" -\u003e \"Windsor, ON, Canada\" (43.46430180,-80.52042120 -\u003e 42.31486680,-83.03656800), duration\u003d2 hours 55 mins, distance\u003d291 km: 18 steps]\",\"[DirectionsLeg: \"Windsor, ON, Canada\" -\u003e \"Montreal, QC, Canada\" (42.31486680,-83.03656800 -\u003e 45.50171230,-73.56721840), duration\u003d8 hours 42 mins, distance\u003d897 km: 21 steps]\"]";
+  private static final String MONTREAL_TO_WATERLOO_NO_WAYPOINTS_JSON =
+      "[\"[DirectionsLeg: \"Montreal, QC, Canada\" -\u003e \"Waterloo, ON, Canada\" (45.50171230,-73.56721840 -\u003e 43.46430180,-80.52042120), duration\u003d6 hours 11 mins, distance\u003d638 km: 22 steps]\"]";
```

Comment:
I wonder if you could simplify this a bit. Either by using a real route that is shorter (produces short strings), or coming up with a fake one. It is hard to understand the current data.
Though, I am not too concerned about this. So if you think that would take too much time, you can leave it as is.

Reply:
I think this could also be changed such that only the content is tested for instead of every JSON detail i.e. `{ } ,`. Since the MVP demo is next Tuesday, I would hold off on this for now. I have created an issue #81 for this!

Reply:
Sounds good

Reply:
Edit: Since the tests now parse the JSON instead https://github.com/googleinterns/step7-2020/pull/63#discussion_r456495939, I thought it would be convenient to have this fixed at the same time. I have replaced the places with arbitrary `A`, `B`, etc. and trimmed off or rounded all the numbers like durations, decimal points and numbers with 100th place digit.

Reply:
`RESOLVED`